lib/src/main/kotlin/org/walletconnect/types/TypeMapConversion.kt

```diff
@@ -78,6 +79,7 @@ fun Map<*, *>.extractError(): Session.Error {
 fun Map<String, *>.getId(): Long =
         (this["id"] as? Double)?.toLong() ?: throw IllegalArgumentException("id missing")
 
+fun Map<*, *>.getChainId(): Long? = (this["chainId"] as? Double)?.toLong()
```

Comment:
I think we can go with `Map<String, *>`

Reply:
Most probably we can't because we're casting `data` object to `Map<*, *>` on `line 57`
```kt
val data = params.firstOrNull() as? Map<*, *> ?: throw IllegalArgumentException("Invalid params")
```

Reply:
I'd change `extractPeerMeta` function as well, since in the function it checks keys as `String`, but yeah. We can go with < * , *> then